code/modules/reagents/chemistry/holder.dm

```diff
@@ -472,10 +472,11 @@ var/const/INJECT = 5 //injection
 
 	return 0
 
-/datum/reagents/proc/clear_reagents()
+/datum/reagents/proc/clear_reagents(exclude_id)
 	for(var/reagent in reagent_list)
 		var/datum/reagent/R = reagent
-		del_reagent(R.id)
+		if(R.id != exclude_id)
+			del_reagent(R.id)
```

Comment:
Yes, actually I have an idea: don't fix slime jelly in this PR, and let me fix the clusterfuck that is our way of handling blood, which will fix the slime jelly issue.